### DISCUSSION TITLE
Add filter capability

### DISCUSSION THREAD

**Discussion by ChefDaddyJay** [jdwilkin4/fcc-forum-clone#51] [Ideas]
Similar to sorting, search algorithms can make for very interesting discussion. Again, though, this is a small project, so perhaps something more like a filtering process would be better. Things we could allow the user to filter by:

- [ ] Min/max replies
- [ ] Min/max views
- [ ] Time range (only show most recent 20, 15, etc.)
- [ ] Filter by category
- [ ] Any others?

**Comment by jdwilkin4** [comment]
I like the filter idea .

Do you have some sort of sketch or figma mockup to add to your post so we can get an idea design wise what you are thinking?

**Reply by ChefDaddyJay** [reply]
I'm gonna have to take some time to figure out Figma, but here's a rough example of what I'm thinking:

![forum-clone-filter-example-1](https://user-images.githubusercontent.com/97722492/202872406-b3aac777-6fdc-4cc2-a38a-116442e923cd.jpg)

![forum-clone-filter-example-2](https://user-images.githubusercontent.com/97722492/202911176-72c7f5aa-b136-43d8-bda2-5a8c398370ba.jpg)

An alternate design might replace the drop-down with radio buttons.

![forum-clone-filter-example-3](https://user-images.githubusercontent.com/97722492/202911796-5073bfad-a0f3-49f4-b8a7-0a678e19fb52.jpg)

**Comment by Dorfieeee** [comment]
I must say that I am not a big fan of this feature for this project.
I would rather see this in a project with big data.

**Reply by ChefDaddyJay** [reply]
Well, sure. Maybe we can just keep it simpler and only filter by category? We'll already be able to sort by the rest anyway. Just have a drop-down to select the category, then only display the most recent posts from that category. It would be nice to have more to work with, but we are limited by the data we're being given. For learning/demonstration purposes, though, the concept is still the same.

**Reply by Dorfieeee** [reply]
We could have a list of categories as buttons above the table. 
- consisting only of the categories that are actually present
- or listing them all
- or a drop-down list

**Reply by ChefDaddyJay** [reply]
I like the buttons idea. Each button could be the same color as its category and display the number of posts in that category, as well. If we list them all, we could just disable the ones that have no posts. That way the categories all get displayed.

**Reply by Dorfieeee** [reply]
Now it's time to design it and split the problem into small issues like I'm trying to do with the sorting process.
